Tests/PersonalityInsightsV3Tests/PersonalityInsightsTests.swift

```diff
@@ -0,0 +1,170 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+import XCTest
+import PersonalityInsightsV3
+
+class PersonalityInsightsTests: XCTestCase {
+
+    private var personalityInsightsV3: PersonalityInsights!
+    private var mobyDickIntro: String!
+    private var kennedySpeech: String!
+    private let timeout: TimeInterval = 5.0
+    private var version: String = "2016-10-20"
+    
+    static var allTests : [(String, (PersonalityInsightsTests) -> () throws -> Void)] {
+        return [
+            ("testProfile", testProfile),
+            ("testContentItem", testContentItem),
+            ("testProfileWithShortText", testProfileWithShortText)
+        ]
+    }
+
+    // MARK: - Test Configuration
+
+    /** Set up for each test by loading text files and instantiating the service. */
+    override func setUp() {
+        super.setUp()
+        continueAfterFailure = false
+        instantiatePersonalityInsights()
+        loadMobyDickIntro()
+        loadKennedySpeech()
+    }
+
+    /** Instantiate Personality Insights. */
+    func instantiatePersonalityInsights() {
+        let username = Credentials.PersonalityInsightsV3Username
+        let password = Credentials.PersonalityInsightsV3Password
+        personalityInsightsV3 = PersonalityInsights(username: username, password: password)
+    }
+    
+    /** Load "MobyDickIntro.txt". */
+    func loadMobyDickIntro() {
+        let bundle = Bundle(for: type(of: self))
+        guard let file = bundle.path(forResource: "MobyDickIntro", ofType: "txt") else {
+            XCTFail("Unable to locate MobyDickIntro.txt file.")
+            return
+        }
+
+        mobyDickIntro = try? String(contentsOfFile: file)
+        guard mobyDickIntro != nil else {
+            XCTFail("Unable to read MobyDickIntro.txt file.")
+            return
+        }
+    }
+
+    /** Load "KennedySpeech.txt." */
+    func loadKennedySpeech() {
+        let bundle = Bundle(for: type(of: self))
+        guard let file = bundle.path(forResource: "KennedySpeech", ofType: "txt") else {
+            XCTFail("Unable to locate KennedySpeech.txt file.")
+            return
+        }
+
+        kennedySpeech = try? String(contentsOfFile: file)
+        guard kennedySpeech != nil else {
+            XCTFail("Unable to read KennedySpeech.txt file.")
+            return
+        }
+    }
+
+    /** Fail false negatives. */
+    func failWithError(error: Error) {
+        XCTFail("Positive test failed with error: \(error)")
+    }
+
+    /** Fail false positives. */
+    func failWithResult<T>(result: T) {
+        XCTFail("Negative test returned a result.")
+    }
+
+    /** Wait for expectations. */
+    func waitForExpectations() {
+        waitForExpectations(timeout: timeout) { error in
+            XCTAssertNil(error, "Timeout")
+        }
+    }
```

Comment:
Let me know if you have any questions about the `failWithError(error:)`, `failWithResult(result:)` or `waitForExpectations()` functions! They can be a bit confusing the first time you see them.